Automattic/wp-calypso
# Focused Launch: Add DomainPicker to DomainDetail view

#### Changes proposed in this Pull Request

* Add the `<DomainPicker />` component from `@automattic/domain-picker` to the `<DomainStep />` in Focused Launch
* Added `jest.config.js` to package for component tests with `@testing-library/react`
* Includes basic component tests for `<DomainStep />`
* Moved flow id to `constants.ts` into a constant named `FOCUSED_LAUNCH_FLOW_ID`

#### Testing instructions

#### Prepare the sandboxed environment:
* Checkout this branch
* Make sure you have an active connection to your sandbox, and that your sandbox is clean and up to date
* In three separate terminal windows, from the root of the project, run:
  - `yarn start`
  - `cd apps/editing-toolkit && yarn dev --sync`
  - `cd apps/wpcom-block-editor && yarn dev --sync`
* Sandbox an _unlaunched_ WordPress site created via the `/start` site creation flow by adding it to your host file

#### Test for changes
* Visit `/page/SANDBOXED_SITE_ID_CREATED_WITH_START.wordpress.com/home?flags=create/focused-launch-flow`
* Click on "Launch" button — Focused Launch modal should appear
* Click on the "View all domains" link — A new placeholder view for the domain details is shown
* You should be presented with the domain picker view.
  - Domain rows should have a "Select" button on them.
  - You should have a row separating your free static sub-domain from the paid domains.
  - Changing the input value should return new premium domains.
  - Clicking "View more results" should reveal a longer list of domains.
  - Clicking "Select" on a chosen domain should change the state of that domain to "Selected".
  - Clicking "Go back" should return you to the initial screen where your selected domain is highlighted in the appropriate summary section.

#### Unit tests
* From the project root, open terminal and run `npm run test-packages`

<img width="1493" alt="Screenshot 2020-11-11 at 14 50 43" src="https://user-images.githubusercontent.com/8639742/98826184-50a43880-242d-11eb-9b6a-aeaf2fbb1452.png">

Fixes https://github.com/Automattic/wp-calypso/issues/46866
